common/tree/tx_graph.go

```diff
@@ -0,0 +1,330 @@
+package tree
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/btcsuite/btcd/btcutil/psbt"
+)
+
+// TxGraph is the reprensation of a directed graph of psbt packets
+// it is used to represent the batch tree
+type TxGraph struct {
+	Root     *psbt.Packet
+	Children map[uint32]*TxGraph // output index -> child graph
+}
+
+// TxGraphChunk is a chunk of TxGraph
+// it is used to serialize and deserialize the graph because TxGraph is recursive
+// a list of TxGraphChunk can be used to reconstruct the TxGraph
+type TxGraphChunk struct {
+	// Tx is the base64 encoded root PSBT
+	Tx string
+	// Children maps root output index to child txid
+	Children map[uint32]string
+}
```

Comment:
isn't it better to have also the Txid here as field so we can get rid of `TxGraphChunk.TxID()` and spare psbt deserialization?